api/api.go

```diff
@@ -325,41 +303,78 @@ func ListEventUsageForResource(db db.SQLClient) echo.HandlerFunc {
 			return errors.New("missing resource_guid")
 		}
 		return withAuthorizedResources(Many, billableViewName, c, db, `
+			with
+			summed_resources as (
+				select
+					id,
+					pricing_plan_id,
+					min(pricing_plan_component_id) as pricing_plan_component_id,
+					sum((price * 100)::bigint) as price_in_pence
+				from
+					monetized_resources
+				where
+					guid = $1
+				group by
+					id, pricing_plan_id
+			)
 			select
-				guid,
-				org_guid,
-				space_guid,
-				pricing_plan_id,
-				pricing_plan_name,
-				iso8601(lower(duration)) as from,
-				iso8601(upper(duration)) as to,
-				(price * 100)::bigint as price_in_pence
+				mr.guid,
+				mr.org_guid,
+				mr.space_guid,
+				mr.pricing_plan_id,
+				mr.pricing_plan_name,
+				iso8601(lower(mr.duration)) as from,
+				iso8601(upper(mr.duration)) as to,
+				sr.price_in_pence::bigint
 			from
-				monetized_resources
-			where
-				guid = $1
+				summed_resources sr
+			inner join
+				monetized_resources mr
```

Comment:
These queries basically joining on to another version of themselves looks a bit odd to me ... it took me quite while to grok it.

could we not simplify them to remove the join something like this:

```
...
                return withAuthorizedResources(Many, billableViewName, c, db, `
                        select
                                guid,
                                org_guid,
                                space_guid,
                                pricing_plan_id,
                                pricing_plan_name,
                                iso8601(lower(duration)) as from,
                                iso8601(upper(duration)) as to,
                                sum((price * 100)::bigint) as price_in_pence
                        from
                                monetized_resources
                        where
                                guid = $1
                        group by
                                guid, org_guid, space_guid, pricing_plan_id, pricing_plan_name, duration
                        order by
                                pricing_plan_id
                `, resourceGUID)
...
```

Reply:
We could but I wanted to avoid it as it looks hacky and it's not clear anymore what are the real group by fields. If you think that's still better/clearer in the end I'm happy to change them.

Reply:
I added an extra commit to change these two queries.

Reply:
nice ... yeah I think that's much clearer.